auslib/blobs/base.py

```diff
@@ -59,13 +64,20 @@ def __init__(self, *args, **kwargs):
         logger_name = "{0}.{1}".format(self.__class__.__module__, self.__class__.__name__)
         self.__class__.log = logging.getLogger(logger_name)
 
-    def isValid(self):
-        """Decides whether or not this blob is valid based."""
+    def validate(self):
+        """Raises a BlobValidationError if the blob is invalid."""
         self.log.debug('Validating blob %s' % self)
-        schema = self.getSchema()
-        # raises exception if broken
-        jsonschema.validate(self, schema)
-        return True
+        validator = jsonschema.Draft4Validator(self.getSchema())
+        # Normal usage is to use .validate(), but errors raised by it return
+        # a massive error message that includes the entire blob, which is way
+        # too big to be useful in the UI. Instead, we iterate over the
+        # individual errors (which are all single sentences which contain
+        # the name of the failing property and why it failed), and return those.
+        errors = [e.message for e in validator.iter_errors(self)]
+        if errors:
+            raise BlobValidationError("Invalid blob! See 'errors' for details.", errors)
+        else:
+            return True
```

Comment:
This return looks artificial to me, it's made for tests only... Maybe delete the `else` block and use `self.assertIsNone(self.blob.validate())` instead? Unfortunately there is no opposite of `this.assertRaises()`